javascript/require-and-import-mini-lesson.md

```diff
@@ -0,0 +1,351 @@
+
+In that perspective, when using Node, if you want to include modules that are in another file, you can use the `require` built-in function. It will first read a JavaScript file, then execute it, and finally return the `exports` object. Ultimately, the `require` function will look for the specified file in the following order:
+
+1. Built-in core Node.js modules. For example: `fs`
+
+2. Installed NPM Modules. It will look in the node_modules folder. For example: `express`
+
+3. Local Modules. 
+
+It is important to note that if the specified module name has a `./`, a `/` or a `../`, it will look for the directory/file in the given path. Please note that it will match the `.js`, `.json`, `.mjs`, `.cjs`, `.wasm` and `*.node` extensions.
+
+Here are some examples of popular Node built-in modules: `HTTP`, `URL`, `path` and `os`. For a more complete list, see this nicely laid out [W3School Reference](https://www.w3schools.com/nodejs/ref_modules.asp).
+
+Now, in the [guided practice below](#guided-practice), you will be able to test the `require` function.
+
+#### What is import?
+
+The `import` statement was first introduced in ES6, in 2015. It is a *static* import statement, which basically means that it will only bring in what has been exclusively and specifically exported from another module.
+
+The syntax is also quite different, as the module path must be a string literal, whereas when we use `require`, the module path can be dynamic.
+
+For example, with `require`, this will work:
+
+```javascript
+const name = "module2"; 
+const obj = require("./" + name);
+```
+
+However, due to the static aspect of `import`, you cannot do:
+
+```javascript
+// THIS WILL NOT WORK
+const name = "module2"; 
+import * from "./" + name;
+```
+
+You can also import multiple functions with the `import` statement. These are called **named imports**. To do so, you simply include the functions in curly braces.
+
+For example:
+
+```javascript
+// Named imports
+import {firstMessage, secondMessage} from './my-module.mjs';
+```
+
+You can also import a complete namespace. To do so, you simply use the following syntax:
+
+```javascript
+// Namespace import
+import * as myModule from './my-module.mjs';
+```
+
+If you simply import the name of the module, you will get the default export. This is common in libraries such as React:
+
+```javascript
+// Default import
+import myModule from './my-module.mjs';
+```
+
+And, of course, you can combine these:
+
+```javascript
+// Combinations:
+import myModule, {firstMessage, secondMessage} from './my-module.mjs';
+```
+
+Finally, you can also refer to an ES Module (if you need more information on modules, see the reference in the [supplement materials](supplement-materials)) This means that you can export objects and methods you have created.
+
+For example:
+
+```javascript
+module.exports = 'This is a big export!';
+```
+
+As with `require`, in the [guided practice](#guided-practice), you will be able to test all of this.
```

Comment:
This is a mix of the two styles, which this article is trying to explain the differences.

Please stick to `export` and `export default` here.

Reply:
yes, thanks,

[c89273b](https://github.com/Techtonica/curriculum/pull/1414/commits/c89273b844a10c1f67a1d0886a773f18a1c327e8)